File: archive/repos/original_repositories/ReconX/scripts/analyzer.js
const fs = require('fs');
const path = require('path');
const crypto = require('crypto');

const ALL_FILES_DIR = "e:/ReconX/allfiles";
const OUTPUT_DIR = "e:/ReconX/intelligence";

const CATEGORIES = {
    "Recon": ["subdomain", "asn", "dns", "crawl", "asset", "amass", "subfinder", "httpx"],
    "Web Scanning": ["nuclei", "dalfox", "fuzz", "dirb", "gobuster", "ffuf", "sqlmap"],
    "Cloud Security": ["aws", "azure", "gcp", "kubernetes", "docker", "s3", "boto3"],
    "API Security": ["graphql", "openapi", "swagger", "rest", "api"],
    "OSINT": ["phoneinfoga", "holehe", "social", "email", "osint"],
    "Exploitation": ["sqli", "ssrf", "xss", "bypass", "privesc", "exploit", "cve"],
    "AI Systems": ["agent", "planner", "reasoning", "llm", "openai", "anthropic", "langchain", "prompt"],
    "Reporting": ["report", "dashboard", "summary", "html", "pdf", "markdown", "jinja"],
    "Orchestration": ["workflow", "scheduler", "queue", "pipeline", "celery", "rabbitmq", "kafka", "airflow", "dag"],
    "Infrastructure": ["dockerfile", "runtime", "deploy", "config", "env", "terraform", "ansible"],
    "Plugin Systems": ["plugin", "registry", "adapter", "module", "extension"],
    "CTF Automation": ["ctf", "challenge", "solver", "pwn"],
    "Burp/MCP Integrations": ["burp", "mcp", "extension", "proxy"],
    "Intelligence/Correlation": ["graph", "relation", "risk", "score", "neo4j", "correlation"]
};

function getHash(content) {
    const normalized = content.replace(/\s+/g, '');
    return crypto.createHash('sha256').update(normalized, 'utf8').digest('hex');
}

function walkSync(dir, filelist) {
    let files = [];
    try {
        files = fs.readdirSync(dir);
    } catch (err) {
        return filelist;
    }
    
    filelist = filelist || [];
    files.forEach(function(file) {
        let filepath = path.join(dir, file);
        try {
            if (fs.statSync(filepath).isDirectory()) {
                filelist = walkSync(filepath, filelist);
            } else {
                filelist.push(filepath);
            }
        } catch (e) {}
    });
    return filelist;
}

function analyze() {
    if (!fs.existsSync(OUTPUT_DIR)){
        fs.mkdirSync(OUTPUT_DIR, { recursive: true });
    }

    let tool_inventory = [];
    let capability_matrix = { "Uncategorized": [] };
    for (let cat in CATEGORIES) {
        capability_matrix[cat] = [];
    }

    let file_hashes = {};
    let dependency_graph = { nodes: [], edges: [] };
    let execution_entrypoints = [];

    let allFiles = walkSync(ALL_FILES_DIR);
    let total_files = 0;

    allFiles.forEach(filepath => {
        total_files++;
        let filename = path.basename(filepath);
        let ext = path.extname(filename).toLowerCase();
        
        let content = '';
        try {
            content = fs.readFileSync(filepath, 'utf8');
        } catch (e) {
            return;
        }

        let h = getHash(content);
        if (!file_hashes[h]) file_hashes[h] = [];
        file_hashes[h].push(filepath);

        let content_lower = content.toLowerCase();
        let name_lower = filename.toLowerCase();

        let matched_categories = [];
        for (let cat in CATEGORIES) {
            for (let kw of CATEGORIES[cat]) {
                if (content_lower.includes(kw) || name_lower.includes(kw)) {
                    matched_categories.push(cat);
                    break;
                }
            }
        }

        if (matched_categories.length === 0) {
            matched_categories.push("Uncategorized");
        }

        let lang = ext ? ext : "unknown";

        let exec_type = "library";
        if (content.includes("if __name__ ==") || content.includes("def main") || content.includes("func main()") || [".sh", ".bash"].includes(ext)) {
            exec_type = "cli_entrypoint";
            execution_entrypoints.push({
                file: filename,
                path: filepath,
                type: "cli",
                language: lang
            });
        } else if (content_lower.includes("flask") || content_lower.includes("fastapi") || content_lower.includes("app.listen")) {
            exec_type = "api_server";
            execution_entrypoints.push({
                file: filename,
                path: filepath,
                type: "api",
                language: lang
            });
        }

        let deps = [];
        if (lang === ".py") {
            let importRegex = /^(?:import|from)\s+([a-zA-Z0-9_\.]+)/gm;
            let match;
            while ((match = importRegex.exec(content)) !== null) {
                deps.push(match[1]);
            }
        } else if (lang === ".go") {
            let importRegex = /"([^"]+)"/g;
            let match;
            while ((match = importRegex.exec(content)) !== null) {
                deps.push(match[1]);
            }
        }
        deps = [...new Set(deps)];

        tool_inventory.push({
            filename: filename,
            path: filepath,
            language: lang,
            purpose: `Matches categories: ${matched_categories.join(', ')}`,
            category: matched_categories,
            dependencies: deps,
            execution_type: exec_type,
            related_tools: []
        });

        matched_categories.forEach(cat => {
            capability_matrix[cat].push(filename);
        });

        if (deps.length > 0) {
            dependency_graph.nodes.push({ id: filename, group: lang });
            deps.forEach(d => {
                dependency_graph.edges.push({ source: filename, target: d });
            });
        }
    });

    let duplicate_map = [];
    for (let h in file_hashes) {
        if (file_hashes[h].length > 1) {
            duplicate_map.push({
                hash: h,
                count: file_hashes[h].length,
                files: file_hashes[h]
            });
        }
    }

    console.log(`Processed ${total_files} files.`);
    console.log(`Found ${duplicate_map.length} duplicate groups.`);

    fs.writeFileSync(path.join(OUTPUT_DIR, "tool_inventory.json"), JSON.stringify(tool_inventory, null, 2));
    fs.writeFileSync(path.join(OUTPUT_DIR, "capability_matrix.json"), JSON.stringify(capability_matrix, null, 2));
    fs.writeFileSync(path.join(OUTPUT_DIR, "duplicate_map.json"), JSON.stringify(duplicate_map, null, 2));
    fs.writeFileSync(path.join(OUTPUT_DIR, "dependency_graph.json"), JSON.stringify(dependency_graph, null, 2));
    fs.writeFileSync(path.join(OUTPUT_DIR, "execution_entrypoints.json"), JSON.stringify(execution_entrypoints, null, 2));

    console.log("Successfully generated JSON files in", OUTPUT_DIR);
}

analyze();

File: archive/repos/original_repositories/ReconX/scripts/analyzer.py
import os
import re
import json
import hashlib
from collections import defaultdict

ALL_FILES_DIR = "e:/ReconX/allfiles"
OUTPUT_DIR = "e:/ReconX/intelligence"

CATEGORIES = {
    "Recon": ["subdomain", "asn", "dns", "crawl", "asset", "amass", "subfinder", "httpx"],
    "Web Scanning": ["nuclei", "dalfox", "fuzz", "dirb", "gobuster", "ffuf", "sqlmap"],
    "Cloud Security": ["aws", "azure", "gcp", "kubernetes", "docker", "s3", "boto3"],
    "API Security": ["graphql", "openapi", "swagger", "rest", "api"],
    "OSINT": ["phoneinfoga", "holehe", "social", "email", "osint"],
    "Exploitation": ["sqli", "ssrf", "xss", "bypass", "privesc", "exploit", "cve"],
    "AI Systems": ["agent", "planner", "reasoning", "llm", "openai", "anthropic", "langchain", "prompt"],
    "Reporting": ["report", "dashboard", "summary", "html", "pdf", "markdown", "jinja"],
    "Orchestration": ["workflow", "scheduler", "queue", "pipeline", "celery", "rabbitmq", "kafka", "airflow", "dag"],
    "Infrastructure": ["dockerfile", "runtime", "deploy", "config", "env", "terraform", "ansible"],
    "Plugin Systems": ["plugin", "registry", "adapter", "module", "extension"],
    "CTF Automation": ["ctf", "challenge", "solver", "pwn"],
    "Burp/MCP Integrations": ["burp", "mcp", "extension", "proxy"],
    "Intelligence/Correlation": ["graph", "relation", "risk", "score", "neo4j", "correlation"]
}

def get_hash(content: str) -> str:
    # simple hash ignoring whitespaces for near-duplicates
    normalized = re.sub(r'\s+', '', content)
    return hashlib.sha256(normalized.encode('utf-8')).hexdigest()

def analyze():
    if not os.path.exists(OUTPUT_DIR):
        os.makedirs(OUTPUT_DIR, exist_ok=True)
        
    tool_inventory = []
    capability_matrix = {cat: [] for cat in CATEGORIES.keys()}
    capability_matrix["Uncategorized"] = []
    
    file_hashes = defaultdict(list)
    dependency_graph = {
        "nodes": [],
        "edges": []
    }
    execution_entrypoints = []
    
    total_files = 0
    for root, _, files in os.walk(ALL_FILES_DIR):
        for file in files:
            total_files += 1
            path = os.path.join(root, file)
            ext = os.path.splitext(file)[1].lower()
            
            try:
                with open(path, 'r', encoding='utf-8', errors='ignore') as f:
                    content = f.read()
            except Exception:
                continue
            
            # Hash
            h = get_hash(content)
            file_hashes[h].append(path)
            
            content_lower = content.lower()
            name_lower = file.lower()
            
            # Determine Categories
            matched_categories = []
            for cat, keywords in CATEGORIES.items():
                for kw in keywords:
                    if kw in content_lower or kw in name_lower:
                        matched_categories.append(cat)
                        break
            
            if not matched_categories:
                matched_categories = ["Uncategorized"]
                
            # Determine Language
            lang = ext if ext else "unknown"
            
            # Execution Role / Entrypoint
            exec_type = "library"
            if "if __name__ ==" in content or "def main" in content or "func main()" in content or ext in [".sh", ".bash"]:
                exec_type = "cli_entrypoint"
                execution_entrypoints.append({
                    "file": file,
                    "path": path,
                    "type": "cli",
                    "language": lang
                })
            elif "flask" in content_lower or "fastapi" in content_lower or "app.listen" in content_lower:
                exec_type = "api_server"
                execution_entrypoints.append({
                    "file": file,
                    "path": path,
                    "type": "api",
                    "language": lang
                })
                
            # Quick Dependencies (just python imports for now to show concept)
            deps = []
            if lang == ".py":
                imports = re.findall(r'^import (\S+)|^from (\S+) import', content, re.MULTILINE)
                for imp in imports:
                    deps.append(imp[0] or imp[1])
            elif lang == ".go":
                imports = re.findall(r'"([^"]+)"', content)
                deps.extend(imports)
                
            deps = list(set(deps))
            
            # Add to inventory
            tool_entry = {
                "filename": file,
                "path": path,
                "language": lang,
                "purpose": f"Matches categories: {', '.join(matched_categories)}",
                "category": matched_categories,
                "dependencies": deps,
                "execution_type": exec_type,
                "related_tools": [] # to be filled
            }
            tool_inventory.append(tool_entry)
            
            for cat in matched_categories:
                capability_matrix[cat].append(file)
                
            # Add deps
            if deps:
                dependency_graph["nodes"].append({"id": file, "group": lang})
                for d in deps:
                    dependency_graph["edges"].append({"source": file, "target": d})
                    
    # Format duplicate map
    duplicate_map = []
    for h, paths in file_hashes.items():
        if len(paths) > 1:
            duplicate_map.append({
                "hash": h,
                "count": len(paths),
                "files": paths
            })
            
    print(f"Processed {total_files} files.")
    print(f"Found {len(duplicate_map)} duplicate groups.")
    
    with open(os.path.join(OUTPUT_DIR, "tool_inventory.json"), 'w') as f:
        json.dump(tool_inventory, f, indent=2)
        
    with open(os.path.join(OUTPUT_DIR, "capability_matrix.json"), 'w') as f:
        json.dump(capability_matrix, f, indent=2)
        
    with open(os.path.join(OUTPUT_DIR, "duplicate_map.json"), 'w') as f:
        json.dump(duplicate_map, f, indent=2)
        
    with open(os.path.join(OUTPUT_DIR, "dependency_graph.json"), 'w') as f:
        json.dump(dependency_graph, f, indent=2)
        
    with open(os.path.join(OUTPUT_DIR, "execution_entrypoints.json"), 'w') as f:
        json.dump(execution_entrypoints, f, indent=2)
        
    print("Successfully generated JSON files in", OUTPUT_DIR)

if __name__ == "__main__":
    analyze()
